nio/src/main/scala/zio/nio/channels/DatagramChannel.scala

```diff
@@ -1,40 +1,146 @@
 package zio.nio.channels
 
 import java.io.IOException
+import java.net.{ DatagramSocket => JDatagramSocket, SocketAddress => JSocketAddress }
 import java.nio.channels.{ DatagramChannel => JDatagramChannel }
-import java.net.{ DatagramSocket => JDatagramSocket }
 
-import zio.nio.core.{ ByteBuffer, SocketAddress }
 import zio.{ IO, Managed, UIO }
+import zio.nio.core.{ ByteBuffer, SocketAddress, SocketOption }
 
+/**
+ * A [[java.nio.channels.DatagramChannel]] wrapper allowing for idiomatic [[zio.ZIO]] interoperability.
+ */
 final class DatagramChannel private[channels] (override protected[channels] val channel: JDatagramChannel)
     extends GatheringByteChannel
     with ScatteringByteChannel {
 
-  def bind(local: SocketAddress): IO[IOException, DatagramChannel] =
-    IO.effect(new DatagramChannel(channel.bind(local.jSocketAddress))).refineToOrDie[IOException]
+  /**
+   * Binds this channel's underlying socket to the given local address. Passing `None` binds to an
+   * automatically assigned local address.
+   *
+   * @param local the local address
+   * @return the datagram channel bound to the local address
+   */
+  def bind(local: Option[SocketAddress]): IO[IOException, DatagramChannel] = {
```

Comment:
in this one I would prefer to make those private and create helper methods `bind` and `connect` in companion object. and probably remove apply from companion

Reply:
Can you take another look?